diff --git a/sageattention3_blackwell/setup.py b/sageattention3_blackwell/setup.py
index 98f2f05..be0952b 100644
--- a/sageattention3_blackwell/setup.py
+++ b/sageattention3_blackwell/setup.py
@@ -150,7 +150,7 @@ class CachedWheelsCommand(_bdist_wheel):
 
 setup(
     name=PACKAGE_NAME,
-    version="1.0.0",
+    version="2.2.0",
     packages=find_packages(
         exclude=(
             "build",
